app/src/components/notifications/LNotification.vue

```diff
@@ -1,35 +1,77 @@
 <script setup lang="ts">
-import { ref } from "vue";
+import { ref, type FunctionalComponent } from "vue";
 import { CheckCircleIcon, ExclamationCircleIcon } from "@heroicons/vue/24/outline";
 import { XMarkIcon } from "@heroicons/vue/20/solid";
-import type { Notification } from "@/stores/notification";
+import { type Notification, useNotificationStore } from "@/stores/notification";
 
 type Props = {
     notification: Notification;
 };
 
-defineProps<Props>();
+const props = defineProps<Props>();
 
 const show = ref(true);
+
+const icon = ref<FunctionalComponent>();
+
+const { removeNotification } = useNotificationStore();
+
+if (props.notification.icon) {
+    icon.value = props.notification.icon;
+} else {
+    switch (props.notification.state) {
+        case "success":
+            icon.value = CheckCircleIcon;
+            break;
+        case "error":
+            icon.value = ExclamationCircleIcon;
+            break;
+        case "info":
+            icon.value = ExclamationCircleIcon;
+            break;
+        case "warning":
+            icon.value = ExclamationCircleIcon;
+            break;
```

Comment:
This could also be written as:

```suggestion
        case "error":
        case "info":
        case "warning":
            icon.value = ExclamationCircleIcon;
            break;
```